zerver/actions/message_send.py

```diff
@@ -104,6 +105,17 @@
 from zerver.tornado.django_api import send_event_on_commit
 
 
+def message_is_idempotent(user: UserProfile, local_id: str, queue_id: str) -> bool:
+    sent_message_local_id = f"{user.realm_id}:{user.id}:<{queue_id}>:{local_id}"
+
+    try:
+        IdempotentMessage.objects.create(key=sent_message_local_id)
+    except IntegrityError:
+        return False
```

Comment:
What restrictions do we have on the format of `queue_id` and `local_id`? 

I guess there's also a question of whether there's a need to squash these together, or we should just have the `IdempotentMessage` table have 4 fields and a different constraint structure, rather than squashing them together into a single `key`. It seems like it'd be better to have separate fields?